packages/logging_cloudwatch/amplify_logging_cloudwatch/lib/src/queued_item_store/drift/drift_queued_item_store.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:amplify_db_common_dart/amplify_db_common_dart.dart';
+import 'package:aws_logging_cloudwatch/aws_logging_cloudwatch.dart';
+import 'package:drift/drift.dart';
+
+part 'drift_queued_item_store.g.dart';
+
+/// SQL schema of data stored in DriftDatabase.
+class DriftQueuedItems extends Table {
+  /// Identifies object in the SQL database.
+  IntColumn get id => integer().autoIncrement()();
+
+  /// The string value stored for this object.
+  TextColumn get value => text()();
+
+  /// The timestamp of the item.
+  TextColumn get timestamp => text()();
+}
+
+/// {@template amplify_analytics_pinpoint_dart.drift_queued_item_store}
+/// Drift class for managing stored [DriftQueuedItems].
+/// {@endtemplate}
+@DriftDatabase(tables: [DriftQueuedItems])
+class DriftQueuedItemStore extends _$DriftQueuedItemStore
+    implements QueuedItemStore {
+  /// {@macro amplify_analytics_pinpoint_dart.drift_queued_item_store}
+  factory DriftQueuedItemStore(String storagePath) {
+    final driftQueryExecutor = connect(
+      name: 'logging_cached_logs',
+      path: storagePath,
+    );
+    return DriftQueuedItemStore._(driftQueryExecutor);
+  }
+
+  DriftQueuedItemStore._(super.driftQueryExecutor);
+
+  // Reminder: Bump this number whenever you change or add a table definition.
+  @override
+  int get schemaVersion => 1;
+  int _currentTotalByteSize = 0;
+
+  @override
+  MigrationStrategy get migration {
+    return MigrationStrategy(
+      onCreate: (Migrator m) async {
+        await m.createAll();
+      },
+      onUpgrade: (Migrator m, int from, int to) async {
+        // Note: From schemaVersion 1->2 we renamed DriftJsonStrings to DriftQueuedItems before the GA release
+        await m.createAll();
+      },
+    );
+  }
+
+  @override
+  Future<void> addItem(String value, String timestamp) async {
+    _currentTotalByteSize += utf8.encode(value).length;
+    _currentTotalByteSize += utf8.encode(timestamp).length;
```

Comment:
The encoded timestamp's byte count will always be the same. Hardcode it in a static var.